test/crawler/keyword/keywords_test.exs

```diff
@@ -5,15 +5,93 @@ defmodule Crawler.Keyword.KeywordsTest do
   alias Crawler.Keyword.Schemas.Keyword
   alias CrawlerWorker.Keyword.CrawlerWorker
 
-  describe "list_keywords/1" do
+  describe "list_user_keywords_by_filter_params/1" do
     test "given a valid user id, returns user's keyword list" do
       %{id: user_id} = insert(:user)
       insert(:keyword, user_id: user_id)
 
       %{id: expected_user_id} = insert(:user)
       expected_keyword = insert(:keyword, user_id: expected_user_id)
 
-      assert Keywords.list_keywords(expected_user_id) == [expected_keyword]
+      assert Keywords.list_user_keywords_by_filter_params(expected_user_id) == [expected_keyword]
+    end
+  end
+
+  describe "list_user_keywords_by_filter_params/2" do
+    test "given a valid user id and valid keyword name, returns user's keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "keyword")
+
+      %{id: expected_user_id} = insert(:user)
+      expected_keyword = insert(:keyword, user_id: expected_user_id, name: "phone")
+      insert(:keyword, user_id: expected_user_id, name: "tv")
+
+      assert [%{name: stored_keyword_name}] =
+               Keywords.list_user_keywords_by_filter_params(expected_user_id, %{
+                 "keyword" => expected_keyword.name
+               })
+
+      assert stored_keyword_name == "phone"
+    end
+
+    test "given a part of valid keyword, returns user's keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "phone")
+      insert(:keyword, user_id: user_id, name: "tv")
+
+      %{id: expected_user_id} = insert(:user)
+      insert(:keyword, user_id: expected_user_id, name: "phone")
+      insert(:keyword, user_id: expected_user_id, name: "tv")
+
+      assert [%{name: stored_keyword_name}] =
+               Keywords.list_user_keywords_by_filter_params(expected_user_id, %{
+                 "keyword" => "hon"
+               })
+
+      assert stored_keyword_name == "phone"
+    end
+
+    test "given an invalid keyword, returns empty keyword list" do
+      %{id: user_id} = insert(:user)
+      insert(:keyword, user_id: user_id, name: "keyword")
```

Comment:
This name should be as `something` right, so it could demonstrative the keyword is existing but does not belong into the user

```suggestion
      insert(:keyword, user_id: user_id, name: "something")
```